controllers/account/account_controller.go

```diff
@@ -807,24 +838,54 @@ func (r *AccountReconciler) nonCCSAssignAccountID(reqLogger logr.Logger, current
 	// update account cr with awsAccountID from aws
 	currentAcctInstance.Spec.AwsAccountID = awsAccountID
 
-	// tag account with hive shard name
-	err = TagAccount(awsSetupClient, awsAccountID, r.shardName)
+	// tag account with hive shard name and compliance tags
+	err = TagAccount(awsSetupClient, awsAccountID, r.shardName, r.appCode, r.servicePhase, r.costCenter, r.complianceTagsEnabled)
 	if err != nil {
 		reqLogger.Info("Unable to tag aws account.", "account", currentAcctInstance.Name, "AWSAccountID", awsAccountID, "Error", error.Error(err))
 	}
 
 	return r.accountSpecUpdate(reqLogger, currentAcctInstance)
 }
 
-func TagAccount(awsSetupClient awsclient.Client, awsAccountID string, shardName string) error {
+func TagAccount(awsSetupClient awsclient.Client, awsAccountID string, shardName string, appCode string, servicePhase string, costCenter string, complianceTagsEnabled bool) error {
```

Comment:
What if instead of passing all the business logic values here as a long list of args, we instead use the above suggested `generateAccountTags` func pattern, and have that generate a `map[string]string` of key + value pairs? 

Then the below logic all goes away, and we just iterate through the key values and tag.

In the earlier paths where we hydrate that map of `key:value`, you can handle `""` values there etc as well. This way, if we need to add more tags in the future, we have a nice little framework for doing so.

Reply:
As requested, I have created the genereateAccountTags func, and made it output a map[string]string. This was a great idea that simplified the code. I have tested the most recent changes with @reedcort on a test cluster. We were able to validate that it tagged a new account, handled it correctly to look at an existing account, as well as verified and updated a tag if it was incorrect.